source/simulator/core.cc

```diff
@@ -1320,6 +1381,26 @@ namespace aspect
     return ret;
   }
 
+  /**
+   * Split the set of DoFs (typically locally owned or relevant) in @p whole_set into blocks
+   * given by the @p dofs_per_block structure.
+   */
+  void  split_by_block (std::vector<IndexSet> &partitioned,
```

Comment:
Let's just make the function return the vector, rather than return it via an argument. With C++11, this is going to cost exactly the same.

If you don't want that, return the computed things through the *last* rather than the *first* argument.